File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let min = prices[0];
    let maxProfit = 0;
    for (let i = 1; i < prices.length; i++) {
        min = Math.min(min, prices[i]);
        const profit = prices[i] - min;
        maxProfit = Math.max(maxProfit, profit);
    }
    return maxProfit;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int prefixMin = prices[0];
        int maxProfit = 0;
        for(int i = 1; i < prices.length; i ++) {
            if(prices[i] < prefixMin) prefixMin = prices[i];
            int curProfit = prices[i] - prefixMin;
            maxProfit = Math.max(maxProfit , curProfit);
        }
        return maxProfit;
    }
}